### PR TITLE
Update Purifying Icicle to deal extra vitality damage only against undead

### PR DESCRIPTION
**Pull request by simonward**
Also sort damage partials

### CODE REVIEW

File: packs/spells/purifying-icicle.json [hunk 1]
```diff
@@ -69,7 +58,18 @@
             "value": "60 feet"
         },
         "requirements": "",
-        "rules": [],
+        "rules": [
+            {
+                "damageType": "vitality",
+                "diceNumber": "@spell.level",
```
Comment (stwlam) [inline packs/spells/purifying-icicle.json:64]:
```suggestion
                "diceNumber": "@spell.rank",
```

File: packs/spells/purifying-icicle.json [hunk 2]
```diff
@@ -69,7 +58,18 @@
...
+                "dieSize": "d4",
+                "key": "DamageDice",
+                "predicate": [
+                    "target:mode:undead"
```
Comment (stwlam) [inline packs/spells/purifying-icicle.json:68]:
```suggestion
                    "target:negative-healing"
```

Comment (simonward) [reply packs/spells/purifying-icicle.json:68]: The description says "...if the target is undead...". It's possible to have negative healing but not be undead?

Comment (stwlam) [reply packs/spells/purifying-icicle.json:68]: Are you going to change heal and harm, too?

**Pull request merged**